src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -112,6 +113,31 @@ private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
                 dotnetPath = Path.GetDirectoryName(isWindows ? dotnetPath : realpath(dotnetPath) ?? dotnetPath);
             }
 
+            foreach (string path in GetDotNetBasePaths(workingDirectory, dotnetPath))
+            {
+                returnedSDKs = true;
+                yield return path;
+            }
+
+            if (Process.GetCurrentProcess().ProcessName.Equals("dotnet"))
+            {
+                // We're already in a dotnet process! Try using it.
+                dotnetPath = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
+                foreach (string path in GetDotNetBasePaths(workingDirectory, dotnetPath))
+                {
+                    returnedSDKs = true;
+                    yield return path;
+                }
+            }
+
+            if (!returnedSDKs)
+            {
+                throw new InvalidOperationException("Failed to find an appropriate version of .NET Core MSBuild. Call to hostfxr_resolve_sdk2 failed. There may be more details in stderr.");
```

Comment:
nit: It doesn't look like getting here necessarily implies that `hostfxr_resolve_sdk2` failed. We may have simply found no SDKs.